github-comment-fetcher/src/main/java/com/tzachz/commentcounter/CommentFetcher.java

```diff
@@ -71,4 +74,19 @@ public void setClock(Clock clock) {
                 .collect(Collectors.toList());
     }
 
+    private List<GHRepo> getInterestingRepos(Set<GHRepo> repos) {
+        final Set<String> repoFilter;
+        if (this.repositories.isEmpty()) {
```

Comment:
Not a must - but an idea:
there's some redundancy in creating a "no-op" filter by creating a set of all names. I feel like using Java 8's `Predicate` might help resolving this - perhaps if we create a class member `Predicate<GHRepo> repoFilter`, and instantiate it in constructor according to the input, e.g.:
```java
if (repositories.isEmpty()) this.repoFilter = any -> true;
else this.repoFilter = repo -> repositories.contains(repo.getName());
```

Then this will be unncessary, and we'd be able to use Java's Stream to apply this in `getComments`:
```java
final List<GHRepo> interestingRepos = repos.stream().filter(repoFilter).collect(Collectors.toList());
```

I know most of the code in this repo isn't abiding by these standards (it's 6 years old...), but I wish it would ;)

Reply:
Makes sense.  I'll make this change and repush.

Reply:
Instead of saving a member variable as a `Predicate` I wrote a predicate method.